refactor: prints `json-render-diagnostics` instead of normal error messages from `cargo`/`rustc`

When `c2rust-refactor` runs `cargo`/`rustc`, it prints `json-render-diagnostics` instead of normal error messages, which is super hard to read.

See https://github.com/immunant/c2rust/actions/runs/22406572073/job/64867978569?pr=1620.

```shell
> cargo build --release
> (cd tests/integration && PATH=$OLDPWD/target/release C2RUST_DIR=$OLDPWD ./test.py lua) |& head -c 10000
    Finished release [optimized] target(s) in 0.10s
info: checking requirements(tests/conf.yml)
info: checking requirements(tests/lua/conf.yml)
lua: make(tests/lua/make.sh)...................................... (0:00:00.000138) OK_CACHED
lua: transpile(tests/lua/transpile.gen.sh)........................ (0:00:03.834229) OK
lua: cargo.transpile(tests/lua/cargo.transpile.gen.sh)............ (0:00:00.514064) OK
lua: check.transpile(tests/lua/test.sh)........................... (0:00:00.569431) OK
lua: refactor(tests/lua/refactor.gen.sh).......................... (0:00:17.656259) FAIL
error: could not compile `repo`
{"message":"unresolved import `crate::src::lbaselib::ctype_h::C2RustUnnamed_9`","code":{"code":"E0432","explanation":"An import was unresolved.\n\nErroneous code example:\n\n```compile_fail,E0432\nuse something::Foo; // error: unresolved import `something::Foo`.\n```\n\nIn Rust 2015, paths in `use` statements are relative to the crate root. To\nimport items relative to the current and parent modules, use the `self::` and\n`super::` prefixes, respectively.\n\nIn Rust 2018 or later, paths in `use` statements are relative to the current\nmodule unless they begin with the name of a crate or a literal `crate::`, in\nwhich case they start from the crate root. As in Rust 2015 code, the `self::`\nand `super::` prefixes refer to the current and parent modules respectively.\n\nAlso verify that you didn't misspell the import name and that the import exists\nin the module from where you tried to import it. Example:\n\n```\nuse self::something::Foo; // Ok.\n\nmod something {\n    pub struct Foo;\n}\n# fn main() {}\n```\n\nIf you tried to use a module from an external crate and are using Rust 2015,\nyou may have missed the `extern crate` declaration (which is usually placed in\nthe crate root):\n\n```edition2015\nextern crate core; // Required to use the `core` crate in Rust 2015.\n\nuse core::any;\n# fn main() {}\n```\n\nSince Rust 2018 the `extern crate` declaration is not required and\nyou can instead just `use` it:\n\n```edition2018\nuse core::any; // No extern crate required in Rust 2018.\n# fn main() {}\n```\n"},"level":"error","spans":[{"file_name":"src/lbaselib.rs","byte_start":493,"byte_end":539,"line_start":31,"line_end":31,"column_start":52,"column_end":98,"is_primary":true,"text":[{"text":"pub use crate::src::lbaselib::ctype_h::toupper;use crate::src::lbaselib::ctype_h::C2RustUnnamed_9;pub use crate::stdlib::C2RustUnnamed_9;pub use crate::stdlib::__ctype_b_loc;pub use crate::stdlib::__ctype_toupper_loc;pub use crate::src::lauxlib::luaL_Reg;pub use crate::src::lauxlib::luaL_argerror;pub use crate::src::lauxlib::luaL_checkany;pub use crate::src::lauxlib::luaL_checkinteger;pub use crate::src::lauxlib::luaL_checklstring;pub use crate::src::lauxlib::luaL_checkoption;pub use crate::src::lauxlib::luaL_checkstack;pub use crate::src::lauxlib::luaL_checktype;pub use crate::src::lauxlib::luaL_error;pub use crate::src::lauxlib::luaL_getmetafield;pub use crate::src::lauxlib::luaL_loadbufferx;pub use crate::src::lauxlib::luaL_loadfilex;pub use crate::src::lauxlib::luaL_optinteger;pub use crate::src::lauxlib::luaL_optlstring;pub use crate::src::lauxlib::luaL_setfuncs;pub use crate::src::lauxlib::luaL_tolstring;pub use crate::src::lauxlib::luaL_typeerror;pub use crate::src::lauxlib::luaL_where;pub use crate::src::lauxlib::LUA_GNAME;pub use crate::lua_h::lua_CFunction;pub use crate::lua_h::lua_Integer;pub use crate::lua_h::lua_KContext;pub use crate::lua_h::lua_KFunction;pub use crate::lua_h::lua_Number;pub use crate::lua_h::lua_Reader;pub use crate::src::lstate::lua_State;pub use crate::lua_h::lua_Unsigned;pub use crate::src::lapi::lua_callk;pub use crate::src::lapi::lua_concat;pub use crate::src::lapi::lua_copy;pub use crate::src::lapi::lua_error;pub use crate::src::lapi::lua_gc;pub use crate::src::lapi::lua_geti;pub use crate::src::lapi::lua_getmetatable;pub use crate::src::lapi::lua_gettop;pub use crate::src::lapi::lua_isstring;pub use crate::src::lapi::lua_load;pub use crate::src::lapi::lua_next;pub use crate::src::lapi::lua_pcallk;pub use crate::src::lapi::lua_pushboolean;pub use crate::src::lapi::lua_pushcclosure;pub use crate::src::lapi::lua_pushinteger;pub use crate::src::lapi::lua_pushnil;pub use crate::src::lapi::lua_pushnumber;pub use crate::src::lapi::lua_pushstring;pub use crate::src::lapi::lua_pushvalue;pub use crate::src::lapi::lua_rawequal;pub use crate::src::lapi::lua_rawget;pub use crate::src::lapi::lua_rawgeti;pub use crate::src::lapi::lua_rawlen;pub use crate::src::lapi::lua_rawset;pub use crate::src::lapi::lua_rotate;pub use crate::src::lapi::lua_setfield;pub use crate::src::lapi::lua_setmetatable;pub use crate::src::lapi::lua_settop;pub use crate::src::lapi::lua_setupvalue;pub use crate::src::lapi::lua_stringtonumber;pub use crate::src::lapi::lua_toboolean;pub use crate::src::lapi::lua_tolstring;pub use crate::src::lapi::lua_type;pub use crate::src::lapi::lua_typename;pub use crate::src::lapi::lua_warning;pub use crate::lua_h::LUA_GCCOLLECT_1;pub use crate::lua_h::LUA_GCCOUNT_1;pub use crate::lua_h::LUA_GCCOUNTB_1;pub use crate::lua_h::LUA_GCGEN;pub use crate::lua_h::LUA_GCINC;pub use crate::lua_h::LUA_GCISRUNNING_1;pub use crate::lua_h::LUA_GCRESTART_1;pub use crate::lua_h::LUA_GCSETPAUSE_1;pub use crate::lua_h::LUA_GCSETSTEPMUL_1;pub use crate::lua_h::LUA_GCSTEP_1;pub use crate::lua_h::LUA_GCSTOP_1;pub use crate::lua_h::LUA_MULTRET;pub use crate::lua_h::LUA_OK;pub use crate::lua_h::LUA_REGISTRYINDEX;pub use crate::lua_h::LUA_RIDX_GLOBALS;pub use crate::lua_h::LUA_TFUNCTION;pub use crate::lua_h::LUA_TNIL;pub use crate::lua_h::LUA_TNONE;pub use crate::lua_h::LUA_TNUMBER_1;pub use crate::lua_h::LUA_TSTRING;pub use crate::lua_h::LUA_TTABLE_1;pub use crate::lua_h::LUA_YIELD;","highlight_start":52,"highlight_end":98}],"label":"no `C2RustUnnamed_9` in `src::lbaselib::ctype_h`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror[E0432]\u001b[0m\u001b[0m\u001b[1m: unresolved import `crate::src::lbaselib::ctype_h::C2RustUnnamed_9`\u001b[0m\n\u001b[0m  \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m--> \u001b[0m\u001b[0msrc/lbaselib.rs:31:52\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m|\u001b[0m\n\u001b[0m\u001b[1m\u001b[38;5;12m31\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m|\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m...\u001b[0m\u001b[0mype_h::toupper;use crate::src::lbaselib::ctype_h::C2RustUnnamed_9;pub use crate::stdlib::C2RustUnnamed_9;pub use crate::stdlib::__ctyp\u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m...\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m| \u001b[0m\u001b[0m                      \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9m^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9mno `C2RustUnnamed_9` in `src::lbaselib::ctype_h`\u001b[0m\n\n"}
{"message":"unresolved import `crate::src::ldebug::lopcodes_h`","code":{"code":"E0432","explanation":"An import was unresolved.\n\nErroneous code example:\n\n```compile_fail,E0432\nuse something::Foo; // error: unresolved import `something::Foo`.\n```\n\nIn Rust 2015, paths in `use` statements are relative to the crate root. To\nimport items relative to the current and parent modules, use the `self::` and\n`super::` prefixes, respectively.\n\nIn Rust 2018 or later, paths in `use` statements are relative to the current\nmodule unless they begin with the name of a crate or a literal `crate::`, in\nwhich case they start from the crate root. As in Rust 2015 code, the `self::`\nand `super::` prefixes refer to the current and parent modules respectively.\n\nAlso verify that you didn't misspell the import name and that the import exists\nin the module from where you tried to import it. Example:\n\n```\nuse self::something::Foo; // Ok.\n\nmod something {\n    pub struct Foo;\n}\n# fn main() {}\n```\n\nIf you tried to use a module from an external crate and are using Rust 2015,\nyou may have missed the `extern crate` declaration (which is usually placed in\nthe crate root):\n\n```edition2015\nextern crate core; // Required to use the `core` crate in Rust 2015.\n\nuse core::any;\n# fn main() {}\n```\n\nSince Rust 2018 the `extern crate` declaration is not required and\nyou can instead just `use` it:\n\n```edition2018\nuse core::any; // No extern crate required in Rust 2018.\n# fn main() {}\n```\n"},"level":"error","spans":[{"file_name":"src/ldebug.rs","byte_start":1985,"byte_end":1995,"line_start":42,"line_end":42,"column_start":1362,"column_end":1372,"is_primary":true,"text":[{"text":"pub use crate::llimits_h::l_mem;pub use crate::llimits_h::l_uint32;pub use crate::llimits_h::ls_byte;pub use crate::llimits_h::lu_byte;pub use crate::llimits_h::lu_mem;pub use crate::llimits_h::Instruction;pub use crate::src::lobject::luaO_chunkid;pub use crate::src::lobject::luaO_pushfstring;pub use crate::src::lobject::luaO_pushvfstring;pub use crate::src::lobject::AbsLineInfo;pub use crate::src::lobject::C2RustUnnamed_5;pub use crate::src::lobject::C2RustUnnamed_6;pub use crate::src::lobject::C2RustUnnamed_7;pub use crate::src::lobject::CClosure;pub use crate::src::lobject::Closure;pub use crate::src::lobject::GCObject;pub use crate::src::lobject::LClosure;pub use crate::src::lobject::LocVar;pub use crate::src::lobject::Node;pub use crate::src::lobject::NodeKey;pub use crate::src::lobject::Proto;pub use crate::src::lobject::StackValue;pub use crate::src::lobject::StkId;pub use crate::src::lobject::TString;pub use crate::src::lobject::TValue;pub use crate::src::lobject::Table;pub use crate::src::lobject::UValue;pub use crate::src::lobject::Udata;pub use crate::src::lobject::UpVal;pub use crate::src::lobject::Upvaldesc;pub use crate::src::lobject::Value;pub use crate::src::lobject::LUA_TCCL_1;pub use crate::src::lobjec
```

--- a/c2rust-refactor/src/driver.rs
+++ b/c2rust-refactor/src/driver.rs
@@ -247,7 +247,9 @@ pub fn clone_config(config: &interface::Config) -> interface::Config {
 
 pub fn create_config(args: &[String]) -> interface::Config {
     let matches = rustc_driver::handle_options(args).expect("rustc arg parsing failed");
-    let sopts = rustc_session::config::build_session_options(&matches);
+    let mut sopts = rustc_session::config::build_session_options(&matches);
+    // Print human readable error (the default).
+    sopts.error_format = Default::default();
     let cfg = interface::parse_cfgspecs(matches.opt_strs("cfg"));
     let check_cfg = interface::parse_check_cfg(matches.opt_strs("check-cfg"));
     let sopts = maybe_set_sysroot(sopts, args);

